public/javascripts/case-progress.js

```diff
@@ -78,4 +78,56 @@
     }
   })
 
+  // ---- Case Workflow add hearing outcome START
+
+  const modal = document.querySelector("#add-hearing-outcome-modal");
+
+  if(modal) {
+    const modalCloseButton = modal.getElementsByClassName("modal-close")[0];
+
+    const hearingOutcomeForm = modal.querySelector("#hearing-outcome-form-row");
+
+    const hearingOutcomeTypeSelect = hearingOutcomeForm.getElementsByTagName('select')[0]
+    const sendOutcomeToAdminButton = hearingOutcomeForm.querySelector('#send-outcome-to-admin')
+    const hearingOutcomeError = hearingOutcomeForm.getElementsByClassName('hearing-outcome-modal-error')[0]
+    const targetHearingIdInput = hearingOutcomeForm.querySelector('#targetHearingId')
+
+    sendOutcomeToAdminButton.onclick = (event) => {
+      if (hearingOutcomeTypeSelect.value === 'NOT_SELECTED') {
+        event.preventDefault()
+        hearingOutcomeError.classList.remove('govuk-!-display-none')
+        hearingOutcomeForm.classList.add('govuk-form-group--error')
+      }
+    }
+
+    hearingOutcomeTypeSelect.onchange = (event) => {
+      if (event.value !== 'NOT_SELECTED' && !(hearingOutcomeError.classList.contains('govuk-!-display-none'))) {
+        hearingOutcomeError.classList.add('govuk-!-display-none')
+        hearingOutcomeForm.classList.remove('govuk-form-group--error')
+      }
+    }
+
+    modalCloseButton.onclick = () => {
+      modal.style.display = "none";
+      hearingOutcomeError.classList.add('govuk-!-display-none')
+      hearingOutcomeForm.classList.remove('govuk-form-group--error')
+    }
+
+    const addOutcomeButtons = document.querySelectorAll('.btn-send-hearing-outcome');
+    addOutcomeButtons.forEach(sendOutcomeButton => {
```

Comment:
Not sure it's a good idea to loop through and attach a handler, can you not add a handler higher up the chain and test based on event.target or something?

Reply:
Can you please go through the html and suggest the changes please? Happy to walk you through what I have done.